src/view-utilities.js

```diff
@@ -153,29 +154,79 @@ var viewUtilities = function (cy, options) {
       shiftKeyDown = false;
     }, "keyup");
 
-    var p_start_x, p_start_y, p_end_x, p_end_y;
+    var rect_start_pos_x, rect_start_pos_y, rect_end_pos_x, rect_end_pos_y;
 
-    cy.one('tapstart', function( event){ 
+    cy.one('tapstart', function(event){ 
       if( shiftKeyDown == true){
-        p_start_x = event.position.x;
-        p_start_y = event.position.y;
+        rect_start_pos_x = event.position.x;
+        rect_start_pos_y = event.position.y;
+
       }
     });
 
-    cy.one('tapend', function( event){
-      p_end_x = event.position.x;
-      p_end_y = event.position.y;
-      var zoomLevel = Math.min( cy.width()/ ( Math.abs(p_end_x- p_start_x)), cy.height() / Math.abs( p_end_y - p_start_y));
+    cy.one('tapend', function(event){
+      rect_end_pos_x = event.position.x;
+      rect_end_pos_y = event.position.y;
+      //Find top left of rectangle
+      if( rect_start_pos_x > rect_start_pos_x){
```

Comment:
I think this line will be ` rect_start_pos_x > rect_end_pos_x`.

Moreover, here rect_start_pos_x and rect_start_pos_y values become undefined if the user click to the Marquee Zoom button and then tap to the viewport without pressing to the shift button. You should check whether these values are null or not. If they are null, then this means that user didn't press the shift button so ignore the marquee zoom.